internal/servers/controller/handlers/scopes/scope_service.go

```diff
@@ -318,6 +318,9 @@ func (s Service) updateInRepo(ctx context.Context, parentScope *scopes.ScopeInfo
 	if name := item.GetName(); name != nil {
 		opts = append(opts, iam.WithName(name.GetValue()))
 	}
+	if primaryAuthMethodId := item.GetPrimaryAuthMethodId(); primaryAuthMethodId != nil {
```

Comment:
Consider adding a check in validateUpdateRequest along the lines of:

```
if item.GetPrimaryAuthMethodId() != nil  && !handlers.ValidId(item.GetPrimaryAuthMethodId().GetValue(), password.AuthMethodPrefix, oidc.AuthMethodPrefix) {
  badFields["primary_auth_method_id"] = "Improperly formatted identifier"
}
```

This type of ValidId check (with multiple possible prefixes) might require https://github.com/hashicorp/boundary/pull/977 to be merged.

Reply:
added todo for dependent PR merge.